### PR TITLE
Adds Service as class member to Client builder 

### PR DESCRIPTION
**Pull request by yisraelU**
Service is now a class member in the ClientBuilder
This allows extending the ClientBuilder and utilizing the service in the extension methods

### CODE REVIEW

File: modules/http4s/src/smithy4s/http4s/SimpleProtocolBuilder.scala [hunk 1]
```diff
@@ -91,27 +95,40 @@ abstract class SimpleProtocolBuilder[P](val codecs: CodecAPI)(implicit
...
+      errorOrUri: Either[InvalidUriError, Uri] = Right(
+        uri"http://localhost:8080"
+      )
   ) {
 
     def uri(uri: Uri): ClientBuilder[Alg, Op, F] =
-      new ClientBuilder[Alg, Op, F](this.client, this.service, uri)
+      new ClientBuilder[Alg, Op, F](this.client, this.service, Right(uri))
+
+    def uri(uri: String): ClientBuilder[Alg, Op, F] =
```
Comment (Baccata) [inline modules/http4s/src/smithy4s/http4s/SimpleProtocolBuilder.scala:107]: This change pollutes the API in a way that breaks bincompat, and I don't think it's worth the breakage.

**Comment by yisraelU** [comment]
I removed the last commit and reset it to just add service as val to ClientBuilder

**Review by Baccata — APPROVED** [review]

**Comment by Baccata** [comment]
Can you amend the PR description ?

**Comment by yisraelU** [comment]
Done

**Pull request merged**